src/actions/contact-request/saveContactRequest.ts

```diff
@@ -1,21 +1,67 @@
 'use server'
-import { createContactRequest } from '@/backend/contact-request/contact-request.service'
+import {
+  createContactRequest,
+  deleteContactRequest,
+  findExistingContactRequest,
+} from '@/backend/contact-request/contact-request.service'
 import { ContactFormRequest } from '@/components/ContactForm/schema'
+import {
+  ContactRequestEmailParams,
+  sendContactRequestEmail,
+} from '../mailing/sendContactRequestEmail'
 
-export const saveContactRequest = async (
-  contactFormRequestData: ContactFormRequest,
-) => {
+export const saveContactRequest = async ({
+  message,
+  senderEmail,
+  senderFullName,
+  recipientEmail,
+  subject,
+  profileId,
+}: ContactFormRequest & ContactRequestEmailParams) => {
   try {
-    const createdContactRequest = await createContactRequest(
-      contactFormRequestData,
-    )
+    const existingContactRequest = await findExistingContactRequest({
+      senderEmail: senderEmail,
+      profileId: profileId,
+    })
+    if (existingContactRequest) {
+      throw Error('You already contacted this dev')
+    }
+    const createdContactRequest = await createContactRequest({
+      senderEmail,
+      senderFullName,
+      subject,
+      profileId,
+      message,
+    })
     if (createdContactRequest) {
+      try {
+        await sendContactRequestEmail({
+          senderEmail: senderEmail,
+          senderFullName: senderFullName,
+          recipientEmail: recipientEmail,
+          subject: subject,
+        })
+      } catch (error) {
+        await deleteSavedContactRequest(senderEmail, profileId)
+        throw Error('Failed to send contact request')
+      }
       return createdContactRequest
     } else {
-      return false
+      throw Error('Failed to save contact request.')
     }
   } catch (error) {
     console.error('Failed to save contact request:', error)
-    return false
+    throw Error('Failed to save contact request.')
+  }
+}
+
+export const deleteSavedContactRequest = async (
```

Comment:
**nit:** we don't need this as separate action for now, just import it from service